disnake/interactions/base.py

```diff
@@ -1762,26 +1763,26 @@ def __repr__(self):
         )
 
     def get_with_type(
-        self, key: Snowflake, data_type: OptionType, default: T = None
+        self, key: Snowflake, data_type: Union[OptionType, ComponentType], default: T = None
     ) -> Union[Member, User, Role, InteractionChannel, Message, Attachment, T]:
-        if data_type is OptionType.mentionable:
+        if data_type is OptionType.mentionable or data_type is ComponentType.mentionable_select:
```

Comment:
is this better?
```suggestion
        if data_type in (OptionType.mentionable, ComponentType.mentionable_select):
```

would also be added for all of the other checks below

Reply:
That would work with the current enum implementation, but break with the introduction of #556 since `in` also checks `__eq__` in addition to identity checks:
```py
from disnake import Enum

class OptionType(Enum):
    role = 8

class ComponentType(Enum):
    channel_select = 8

print(OptionType.role is ComponentType.channel_select)  # False
print(OptionType.role == ComponentType.channel_select)  # False
print(OptionType.role in (ComponentType.channel_select,))  # False
```

With the new enums (or just stdlib enums), and using `int, Enum` as the base, you get `False, True, True`.

Either we keep the identity checks here, or avoid `Union[OptionType, ComponentType]` by splitting `InteractionDataResolved` (since that is arguably a bit weird anyway).

Reply:
ah, I see